app/Http/Controllers/MicrotaskingController.php

```diff
@@ -0,0 +1,100 @@
+use Illuminate\Support\Facades\Log;
+
+class MicrotaskingController extends Controller
+{
+    public function index()
+    {
+        if (Auth::check()) {
+            $currentUserId = Auth::user()->id;
+
+            $currentUserQuests = $this->getUserContributions($currentUserId)['quests'];
+            $currentUserContributions = $this->getUserContributions($currentUserId)['contributions'];
+        } else {
+            // TODO: if anon, we display different element.
+            $currentUserQuests = 0;
+            $currentUserContributions = 0;
+        }
+
+        $tag = config('restarters.microtasking.discussion_tag');
+
+        return view('microtasking.dashboard', [
+            'totalContributions' => $this->getTotalContributions(),
+            'currentUserQuests' => $currentUserQuests,
+            'currentUserContributions' => $currentUserContributions,
+            'topics' => $this->getDiscussionTopics($tag, 5),
+            'seeAllTopicsLink' => env('DISCOURSE_URL') . "/tag/{$tag}/l/latest"
+        ]);
+    }
+
+    private function getUserContributions($userId)
+    {
+        $faultCatContributions = DB::select('select count(*) as total from devices_faults_opinions where user_id = :userId', ['userId' => $userId])[0]->total;
+        $miscCatContributions = DB::select('select count(*) as total from devices_misc_opinions where user_id = :userId', ['userId' => $userId])[0]->total;
+        $mobifixContributions = DB::select('select count(*) as total from devices_faults_mobiles_opinions where user_id = :userId', ['userId' => $userId])[0]->total;
+
+        $quests = 0;
+        if ($faultCatContributions > 0) $quests++;
+        if ($miscCatContributions > 0) $quests++;
+        if ($mobifixContributions > 0) $quests++;
+
+        return [
+            'quests' => $quests,
+            'contributions' => $faultCatContributions + $miscCatContributions + $mobifixContributions
+        ];
+    }
+
+    private function getTotalContributions()
+    {
+        $faultCatContributions = DB::select('select count(*) as total from devices_faults_opinions')[0]->total;
+        $miscCatContributions = DB::select('select count(*) as total from devices_misc_opinions')[0]->total;
+        $mobifixContributions = DB::select('select count(*) as total from devices_faults_mobiles_opinions')[0]->total;
+
+        return $faultCatContributions + $miscCatContributions + $mobifixContributions;
+    }
+
+    private function getDiscussionTopics($tag, $numberOfTopics = null)
+    {
+        if (!Auth::check())
+            return [];
+
+        $username = Auth::user()->username;
+        $client = app('discourse-client', ['username' => $username]);
+
+        $topics = [];
+
+        try {
+            $endpoint = "/tag/{$tag}/l/latest.json";
+            $response = $client->request('GET', $endpoint);
+            $discourseResult = json_decode($response->getBody());
+
+            $topics = $discourseResult->topic_list->topics;
```

Comment:
In my test environment, $topics was NULL, which the code doesn't handle.  That might be because I'm not set up right for Discourse testing.  So I can't tests this, but probably worth coding for it too.

Reply:
Hmm, did you get an error?  It should catch the exception, log it, and return an empty array.  (Which yes I think I've forgotten to handle on the client side with a message of 'no topics found').